proposals/lip-0042.md

```diff
@@ -135,11 +139,11 @@ This function is used to retrieve the reward of a block at a given height.
 ##### Execution
 
 ```python
-def getBlockReward(blockHeader: BlockHeader) -> tuple[uint64, uint32]:
+def getBlockReward(blockHeader: BlockHeader, blockAssets: BlockAssets) -> tuple[uint64, uint32]:
   if not Token.userSubstoreExists(blockHeader.generatorAddress, TOKEN_ID_REWARD):
     return (0, REWARD_REDUCTION_NO_ACCOUNT)
 
-  if Random.isSeedRevealValid(blockHeader.generatorAddress, blockHeader.seedReveal) == False:
+  if Random.isSeedRevealValid(blockHeader.generatorAddress, blockAssets) == False:
```

Comment:
Hmm now that I look it again, it seems to be a bit problematic to do this check in `afterTransactionsExecute`, because we don't know the order of execution between random and reward modules. In particular, if the random module's `afterTransactionsExecute` has been already executed and the `validatorReveals` array is updated, then this call of `isSeedRevealValid` might not produce a correct result. 

This could be avoided either by moving this to `beforeTransactionsExecute` or perhaps by adding height as a parameter to this function, so that even if an item with height = b.height is already inserted in the `validatorReveals` array, then it could be somehow ignored.  

I think it makes more sense the first choice and then the two reward modules will also be aligned.

Reply:
I think it would be nicer to move it to `beforeTransactionExecute` (similar to dynamic block rewards) but I will give it a more thought and discuss it again.